pkgs/build-support/fetchbittorrent/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib, runCommand, transmission-withoutSystemd, rqbit, writeScript, formats, cacert, rsync }:
+let
+  urlRegexp = ''.*xt=urn:bt[im]h:([^&]{64}|[^&]{40}).*'';
+in
+{ url
+, name ?
+  if (builtins.match urlRegexp url) == null then
+    "bittorrent"
+  else
+    "bittorrent-" + builtins.head (builtins.match urlRegexp url)
+, config ? if (backend == "transmission") then { } else throw "json config for configuring fetchFromBitorrent only works with the transmission backend"
+, hash
+, backend ? "transmission"
+, recursiveHash ? true
+, postFetch ? ""
+, postUnpack ? ""
+}:
+let
+  afterSuccess = writeScript "kill.sh" ''
+    ${postUnpack}
```

Comment:
What's this for?

Reply:
@roberth In case people want to run something before and after the torrent is complete, i.e postUnpack, as all the other fetchers have.